pydeseq2/dds.py

```diff
@@ -305,6 +315,17 @@ def __init__(
         # Initialize the inference object.
         self.inference = inference or DefaultInference(n_cpus=n_cpus)
 
+    @property
+    def variables(self):
+        """Get the names of the variables used in the model definition."""
+        try:
+            return self.obsm["design_matrix"].model_spec.variables_by_source["data"]
+        except AttributeError:
+            raise ValueError(
+                """Retrieving variables is only possible if the model was initialized
+                using a formula."""
+            ) from None
```

Comment:
Maybe this stuff could really become a Mixin as you suggested, then we can more easily reuse it across pyDESeq2 and pertpy.

Reply:
If it's simpler for `perpty` than yes I can put this in a Mixin.

There's a difference though, because here the design is stored in `.obsm["design_matrix"]` as opposed to `.design` in `pertpy`.

Reply:
I'm not entirely sure yet what's the best solution. Maybe we just leave it as is now, and when I'll look into refactoring pertpy I can propose a PR with changes to pyDESeq2 if required.